src/V3Class.cpp

```diff
@@ -35,8 +35,12 @@ class ClassVisitor : public AstNVisitor {
     AstUser1InUse m_inuser1;
     string m_prefix;  // String prefix to add to name based on hier
     AstScope* m_classScopep = nullptr;  // Package moving scopes into
+    AstScope* m_packageScopep = nullptr;  // Class package scope
+    AstNodeFTask* m_ftaskp = nullptr;  // Current task
     typedef std::vector<std::pair<AstNode*, AstScope*>> MoveVector;
     MoveVector m_moves;
+    typedef std::map<AstNode*, AstVarScope*> VarToScopeMap;
+    VarToScopeMap m_varToScopeMap;  // Map for Var -> VarScope mappings
```

Comment:
A map with a node index can generally use "user#p" instead.  Document this use:

```
    //  AstClass::user1()       -> bool.  True if iterated already
    //  AstVar::user1p()        -> AstVarScope*.  Scope for this variable
```

But are you sure this code works correctly doing this - that is there can be many scopes for one variable, so are you certain that you'll only see a single scope and process it before seeing another?

Reply:
I'm not – I must admit that, so far, scopes have been the most difficult part of Verilator to grasp.
Where could I encounter those other scopes? What if I limited myself to ones that are directly under a class?

Reply:
Actually as you're only under a class you will probably be ok.  Add an assert when you insert that there isn't already a scope for that var, and at least we'll be protected.

I think t_var_static perhaps needs some improvements to have a submodule instantiated twice, to make sure statics are properly handled when the same module is double-instantiated.